### PR TITLE
[#678] xmleditor test language demonstrating FailedPredicateException.

### PR DESCRIPTION
**Pull request by miklossy**
- Add org.eclipse.xtext.ui.tests.xmleditor test language for the fix of
the content assist freeze on FailedPredicateException.
- Add test cases to the tests/org.eclipse.xtext.ui.tests.xmleditor
package.

Signed-off-by: Tamas Miklossy <miklossy@itemis.de>

### CODE REVIEW

**Comment by cdietrich** [comment]
@miklossy it would be nice if you can temporarily change upstream to use core from your branch to see if this is actually working

**Comment by miklossy** [comment]
@cdietrich How exactly should I do that?

**Comment by cdietrich** [comment]
either you get the adjustPipelines script in umbrella running on windows our you simply edit this file and place: https://github.com/eclipse/xtext-eclipse/blob/e4699cfeba5b1e75cb01e6f07c811624247892ea/releng/org.eclipse.xtext.target/org.eclipse.xtext.target-oxygen.target#L35

**Comment by miklossy** [comment]
Thanks for the hints @cdietrich I added a [commit](https://github.com/eclipse/xtext-eclipse/pull/1138/commits/1071ed0c4af34d8faea9ec9abd75bab52db018b3) to the PR that temporary changes the upstream in the target configuration.

File: org.eclipse.xtext.ui.tests/src-gen/org/eclipse/xtext/ui/tests/xmleditor/AbstractXmlRuntimeModule.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+import org.eclipse.xtext.ui.tests.xmleditor.serializer.XmlSemanticSequencer;
+import org.eclipse.xtext.ui.tests.xmleditor.serializer.XmlSyntacticSequencer;
+import org.eclipse.xtext.ui.tests.xmleditor.services.XmlGrammarAccess;
+import org.eclipse.xtext.ui.tests.xmleditor.validation.XmlValidator;
+
+/**
+ * Manual modifications go to {@link XmlRuntimeModule}.
+ */
+@SuppressWarnings("all")
+public abstract class AbstractXmlRuntimeModule extends DefaultRuntimeModule {
```
Comment (szarnekow) [inline org.eclipse.xtext.ui.tests/src-gen/org/eclipse/xtext/ui/tests/xmleditor/AbstractXmlRuntimeModule.java]: Personally I'd prefer XmlTestLanguage, e.g. a name that makes it obvious that this is just for testing purpose. Maybe even prefix it with the bug id.

**Comment by miklossy** [comment]
Thanks for the feedback @szarnekow ! I changed the name of the test language to `Bug678XmlTestLanguage`.

**Comment by miklossy** [comment]
Thanks for the hint, @szarnekow ! I took out the commit regarding the temporar TP configuration from the PR.

**Pull request merged**